numpy/lib/ufunclike.py

```diff
@@ -189,7 +189,8 @@ def isposinf(x, out=None):
     try:
         signbit = ~nx.signbit(x)
     except TypeError as e:
-        raise TypeError('This operation is not supported for complex values '
+        dtype = nx.asanyarray(x).dtype
+        raise TypeError(f'This operation is not supported for {dtype} values '
```

Comment:
The exception message would previously assume that that the dtype of the (invalid) array was complex. This is not necessarily true though, as the very same exception will also be triggered by `timedelta64` and `datetime64` arrays.